packages/components/src/components/TokenGate/TokenGate.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState, ReactNode } from 'react';
+import { ethers } from 'ethers';
+import { Spinner } from '@chakra-ui/react';
+export interface TokenGateProps {
+  /**
+   * The provider or signer to fetch the address from the ens
+   */
+  provider: ethers.providers.Web3Provider;
+  /**
+   * The token contract address
+   */
+  contractAddress: string;
+  /**
+   * The token quantity required to access child component. Default=1
+   */
+  requiredQuantity?: number;
+  /**
+   * Child nodes
+   */
+  children: ReactNode;
+  /**
+   * Optional message if access denied
+   */
+  deniedMessage?: ReactNode;
+  loader?: boolean;
+}
+export const TokenGate: React.FC<TokenGateProps> = ({
+  contractAddress,
+  provider,
+  requiredQuantity = 1,
+  children,
+  deniedMessage,
+  loader = false,
+}) => {
+  const erc20Abi = ['function balanceOf(address owner) view returns (uint256)'];
+  const [tokenQuantity, setTokenQuantity] = useState<number>(0);
+  const [loadedStatus, setloadedStatus] = useState<boolean>(false);
+  // connect to contract address to get balance
+  async function getTokenBalance() {
```

Comment:
u can use the useTokenBalance hook from this library aswell
  const { balance, decimals ,loading,error} = useTokenBalance({ tokenAddress, accountAddress });

Reply:
So i was thinking more about the component a little more. Since it's simply a UI component should it only take the contract and balance as parameters? Then in the core repo wrap it with a provider for token balances? I was hesitant to use one of our hooks for the UI repo. Seemed more appropriate for a core component. Thoughts?

Reply:
@hone1er yeah that's right. We don't want to add our `hooks` package as a dependency to the `components` / UI package.